src/services/storage/secureStorage.ts

```
@@ -0,0 +1,178 @@
+   * only occurs when explicitBiometricPrompt is provided.
+   *
+   * @param key - The key to retrieve
+   * @param options - Retrieval options (biometric prompt)
+   * @returns The stored credentials, or false if not found or authentication failed
+   */
+  getItem: async (
+    key: string,
+    options?: SecureStorageOptions,
+  ): Promise<Keychain.UserCredentials | false> => {
+    try {
+      // If explicit biometric prompt is provided, show it first
+      if (options?.explicitBiometricPrompt) {
+        const hasVerified = await rnBiometrics.simplePrompt({
+          promptMessage: options.explicitBiometricPrompt.title,
+          cancelButtonText: options.explicitBiometricPrompt.cancel,
+        });
+        if (!hasVerified.success) {
+          return false;
+        }
+      }
+
+      const result = await Keychain.getGenericPassword({
+        service: `${DEFAULT_SERVICE}_${key}`,
+        accessControl: SECURE_KEYCHAIN_OPTIONS.accessControl,
+      });
+
+      return result;
+    } catch (error) {
+      logger.error(
+        "secureStorage.getItem",
+        `Error retrieving key from keychain: ${key}`,
+        error,
+      );
+      return false;
+    }
+  },
+
+  /**
+   * Removes one or more items from secure storage
+   *
+   * @param keys - The key(s) to remove
+   */
+  remove: async (keys: string | string[]): Promise<void> => {
+    try {
+      if (Array.isArray(keys)) {
+        await Promise.all(
+          keys.map((key) =>
+            Keychain.resetGenericPassword({
+              service: `${DEFAULT_SERVICE}_${key}`,
+            }),
+          ),
+        );
+        return;
+      }
+
+      await Keychain.resetGenericPassword({
+        service: `${DEFAULT_SERVICE}_${keys}`,
+      });
+    } catch (error) {
+      logger.error(
+        "secureStorage.remove",
+        "Error removing keys from keychain",
+        error,
+      );
+      // Don't throw since removal failures shouldn't block execution
```

would we ever need to return an error response for the UI to provide feedback in this case?